src/message-handler/src/NetworkSerializer.cpp

```diff
@@ -0,0 +1,29 @@
+#include "NetworkSerializer.h"
+#include <pb_encode.h>
+
+NetworkSerializer::NetworkSerializer(INetworkOutputStream& stream, INetworkManager& manager) :
+    m_outputStream(stream), m_networkManager(manager) {}
+
+bool NetworkSerializer::serializeToStream(const MessageDTO& message) {
```

Comment:
I see what you do here, would it be possible to do composition instead of redefining a serializer?
What I mean is that you could instanciate a HiveMindHostSerializer instead of redefining the stream callback and all that. Sot the serializeToStream would be

``` cpp

bool NetworkSerializer::serializeToStream(const MessageDTO& message) {
      char destination[16];
    if (!m_networkManager.getIPFromRobotID(message.getDestinationId(), destination,
                                           sizeof(destination))) {
        return false;
    }

    if (!m_outputStream.setDestination(destination)) {
        return false;
    }
    m_hivemindHostSerializer.serialize(message); // The composition of the serializer, you could build it at construction
}

```

Reply:
good point